Source/carthage/Build.swift

```diff
@@ -128,6 +132,17 @@ public struct BuildCommand: CommandType {
 			return buildProducer
 		} else {
 			let currentProducers = buildInDirectory(directoryURL, withConfiguration: options.configuration, platform: options.buildPlatform.platform)
+				|> catch { error -> SignalProducer<BuildSchemeProducer, CarthageError> in
+					switch error {
+					case let .NoSharedFrameworkSchemes(project, _):
+						// Log that building the current project is being skipped.
+						eventSink.put(.SkippedBuilding(project, error.description))
+						return .empty
+
+					default:
+						return SignalProducer(error: error)
+					}
+				}
```

Comment:
This logic is duplicated here. Can we move it inside `buildInDirectory` instead of repeating it?